lib/types/TokenManager.ts

```diff
@@ -11,16 +11,17 @@ export interface TokenManagerError {
 
 export declare type TokenManagerErrorEventHandler = (error: TokenManagerError) => void;
 export declare type TokenManagerEventHandler = (key: string, token: Token, oldtoken?: Token) => void;
-
+export declare type TokenManagerSetStorageEventHandler = (storage: Tokens) => void;
+export declare type TokenManagerAnyEventHandler = TokenManagerEventHandler | TokenManagerErrorEventHandler | TokenManagerSetStorageEventHandler;
 
 export declare type AccessTokenCallback = (key: string, token: AccessToken) => void;
 export declare type IDTokenCallback = (key: string, token: IDToken) => void;
 export declare type RefreshTokenCallback = (key: string, token: RefreshToken) => void;
 
 // only add methods needed internally
 export interface TokenManagerInterface {
-  on: (event: string, handler: TokenManagerErrorEventHandler | TokenManagerEventHandler, context?: object) => void;
-  off: (event: string, handler?: TokenManagerErrorEventHandler | TokenManagerEventHandler) => void;
+  on: (event: string, handler: TokenManagerAnyEventHandler, context?: object) => void;
```

Comment:
Do you think we could handle this better with function overloads, like we are doing on SIW: https://github.com/okta/okta-signin-widget/blob/okta-signin-widget-6.3.3/src/types/api.ts#L10
Then it will be clear which type of event has which type of callback. As is, I'm not sure as a developer how to implement my handler function - it seems to just have one of 3 shapes, so I have to be prepared for any of the variations. Would be better I think to know which type of callback I should implement for a specific event.